coco/master.py

```diff
@@ -44,25 +45,30 @@ class Master:
     Loads and keeps the config and endpoints. Endpoints are called through this module.
     """
 
-    def __init__(self, config_path):
+    def __init__(self, conf, reset):
         """
         Coco Master.
 
         Parameters
         ----------
-        config_path : os.PathLike
+        conf : os.PathLike
             Path to the config file.
+        reset : bool
+            Whether to reset internal state on start.
         """
 
         # In case constructor crashes before this gets assigned, so that destructor doesn't fail.
         self.qworker = None
-
         self.state = None
 
         # Load the config
-        self._load_config(Path(config_path))
+        self._load_config(Path(conf))
         logger.setLevel(self.config["log_level"])
 
+        if reset is True:
+            # Reset the internal state
+            asyncio.run(self.state.process_post())
+
```

Comment:
If coco stops at the `load_config`, does that mean it will never reach this part of the logic?

Reply:
I should have added more details about the problem when it happened. But `load_config` shouldn't be the problem, as the file that it reads can be edited if faulty. The problem was, I think, that a config lead to a problem in the persistent internal state of coco that would make it crash before one could call the endpoint to reset the state.

Reply:
We want to load the config before doing the reset, because we want to be able to know about paths in the state that are configured to be excluded from reset (like if the cluster is running).

Reply:
Understood!